### ISSUE TITLE
Block Functionality

### ISSUE THREAD

**Issue by uselessSABOTAGE** [samolego/SimpleVillagers#5]
Hello,

I recently installed your mod and am having an issue with the villager blocks. I can pick up villagers fine, but I can't seem to craft the blocks.  I have followed the recipes provided on CurseForge. What happens is nothing will show up in the recipe output, however if I click there the items in the recipe input are consumed and I do *not* receive anything for craft.

**Comment by uselessSABOTAGE** [comment]
Update:

So, the blocks ended up in my inventory and I didn't realize they were there until I tried to completely fill my inventory to move some other items. Turns out they don't show up as an entity in my inventory. However, they did allow me to place them.

**Comment by samolego** [comment]
Huh, that's really strange ... Could you screenshot / record the behaviour?

**Comment by uselessSABOTAGE** [comment]
I can, but probably not for like 12 hours after I get off work.

**Comment by uselessSABOTAGE** [comment]
The video was too big to upload here, so I tried my hand at putting it on youtube, hopefully this link work.  Let me know if it doesn't: https://youtu.be/wxIOi3GMKxE

In the video I pick up a villager, then set him back down to show the mod functions for that aspect. Then I craft the two blocks. Once I craft a block, I move my mouse around each inventory slot to show no tooltips or items appeared. Then, I fill each inventory space with stone, which then shows where the blocks ended up in my inventory. I move those "empty" spaces to my hotbar to place. Then, I place the blocks and break them, again showing no entity on the ground or in my inventory once picked up again. The last thing I do, is show I can place the villager into the blocks and retrieve it. 

As an FYI, the mod is installed both client and server side.

**Comment by samolego** [comment]
Wow, that's ... super strange. I haven't come across anything like that while developing 😅 ...
2 ideas come to my mind; resource pack somehow broke (I saw there's missing translation for villager item as well) or something is up with mod compatibility ...

**Comment by uselessSABOTAGE** [comment]
Indeed, very strange. I have attached two zips, one of my client side mods and the other of my server side mods so you can see what I am working with and if any of that may be the issue.
[Client Mods.zip](https://github.com/samolego/SimpleVillagers/files/8371162/Client.Mods.zip)
[Server Mods.zip](https://github.com/samolego/SimpleVillagers/files/8371164/Server.Mods.zip)

**Comment by samolego** [comment]
Are there any errors in console?
Tested today, simplevillagers by itself doesn't cause any problems.
Try to perform a binary search to find the conflicitng mod combination.

**Comment by uselessSABOTAGE** [comment]
> Are there any errors in console?

I have attached the logs since I installed the mod. I didn't notice anything. 

> Try to perform a binary search to find the conflicitng mod combination.

You're speaking over my head here. I'll try to do research about what you mean, but if you give me some directions or a link with directions, I will do that.  Otherwise, I did provide you with all the mods I run on my server and my clients.  

Let me know what else you might need to sort this. Thanks for your help!
[Logs - 2022-03-27-1.zip](https://github.com/samolego/SimpleVillagers/files/8375661/Logs.-.2022-03-27-1.zip)

**Comment by samolego** [comment]
Binary search is a faster way of finding the conflicting combination;
Remove half of the mods, see if problem persists.
If it does, remove half of current mods (so 1/4 of all) & test again.
If it doesn't, that means that those mods are ok, so use the other half to reproduce the error.
You can repeat that so that you get a minimum required mod combination.

**Comment by samolego** [comment]
There sadly isn't any useful info in the logs :(.

Oh, and try removing the client-*only* mods first, as I suspect that sth. is going on there.

**Comment by uselessSABOTAGE** [comment]
Sorry the logs didn't have any clues.  Thanks for the information about the binary search.  I don't really have time to do that until the weekend.  I will update you as soon as I am able.

**Comment by samolego** [comment]
Ok, no rush :)

**Comment by uselessSABOTAGE** [comment]
So, I finally got the time to try swapping mods.  In single player, I took all the mods off and re-added them one by one, checking between each one and they all worked no issues.  When I tried it on my server, that wasn't the case. I did the same thing, just the villager mod and fabric API - worked. After adding any other mod, it didn't behave as it should. At first, it would use random textures for the villager blocks, then eventually, it would behave as in the video. I honestly don't know what to do.

**Comment by samolego** [comment]
Thanks for information, that could perhaps be connected with polymer ...

**Comment by uselessSABOTAGE** [comment]
You're welcome. If there is anything else you'd like me to try, please let me know.

**Comment by SiskSjet** [comment]
There appears to be an issue when adding a mod that adds blocks or items. I didn't notice this issue in single player.
Sadly, there is nothing useful in logs.

**Comment by tallytarik** [comment]
I was able to fix this by adding Polymer on my client. Didn't need to add it to the server.
